plotly_resampler/figure_resampler/figurewidget_resampler.py

```diff
@@ -56,11 +56,33 @@ def __init__(
         f = self._get_figure_class(go.FigureWidget)()
         f._data_validator.set_uid = False
 
-        if isinstance(figure, BaseFigure):  # go.Figure or go.FigureWidget or AbstractFigureAggregator
-            # A base figure object, we first copy the layout and grid ref
+        if isinstance(figure, BaseFigure):  
+            # A base figure object, can be; 
+            # - a base plotly figure: go.Figure or go.FigureWidget
+            # - a plotly-resampler figure: subclass of AbstractFigureAggregator
+            # => we first copy the layout, grid_str and grid ref
             f.layout = figure.layout
+            f._grid_str = figure._grid_str
             f._grid_ref = figure._grid_ref
             f.add_traces(figure.data)
+        elif isinstance(figure, dict) and (
+            "data" in figure or "layout" in figure # or "frames" in figure  # TODO
```

Comment:
TODO?

Reply:
Well we did not support it before, nor do we support it now. Hence TODO
(wasn't really aware of frames functionality of plotly)

Reply:
should be in FAQ!